R/dependencies.R

```diff
@@ -75,64 +73,43 @@ add_project_to_local_repos <- function(project, local_repos) {
 #'
 #' @examples
 #' \dontrun{
-#' install_deps()
-#'
-#' # install all dependencies
-#' install_deps(direction = c("upstream", "downstream"))
-#'
-#' remove.packages("utils.nest")
-#' install_deps("../scratch1/utils.nest")
+#' install_deps(list(list(
+#' repo = "insightsengineering/utils.nest",
+#' host = "https://github.com"
+#' )), feature = "main")
 #' }
-#'
-install_deps <- function(project = ".", feature = NULL,
-                                  local_repos = get_local_pkgs_from_config(),
-                                  direction = "upstream",
-                                  install_project = TRUE, dry_install = FALSE, verbose = 0,
-                                  install_external_deps = TRUE, ...) {
+install_deps <- function(repos_to_process,
+                         feature,
```

Comment:
Would we leave the default feature = NULL?

Reply:
No here, because the default feature is inferred from the `project` folder which is not provided here.